File: src/types/workday.ts


export interface WorkdayPayload {
    start_user?: string;
    end_user?: string;
    start_time?: string;
    end_time?: string;
    location_id: string;
    total_sales?: number;
    total_taxes?: number;
    total_ticket_count?: number;
    work_period_informations?: string;
    department_ticket_informations?: string;
    add_on?: string;
    auto_closed?: boolean;
    external_processed?: boolean;
    work_period_day?: string;
    business_date?: string;
}

export interface WorkdayResponse {
    status: number;
    wid: string;
}

export interface WorkdayUpdateResponse {
    status: number;
    wid: string;
}

export interface LocalWorkday {
    id: number;
    workdayId?: string;
    startUser?: string;
    endUser?: string;
    startTime?: string;
    endTime?: string;
    locationId: string;
    totalSales?: number;
    totalTaxes?: number;
    totalTicketCount?: number;
    workPeriodInformations?: string;
    departmentTicketInformations?: string;
    addOn?: string;
    autoClosed?: boolean;
    externalProcessed?: boolean;
    workPeriodDay?: string;
    businessDate?: string;
    syncStatus: "PENDING" | "SYNCING" | "SYNCED" | "FAILED";
    syncError?: string;
    createdAt: string;
    updatedAt?: string;
}

File: src-tauri/src/db/models/workday.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Workday {
    pub id: Option<i64>,
    pub workday_id: Option<String>,
    pub start_user: Option<String>,
    pub end_user: Option<String>,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub location_id: String,
    pub total_sales: Option<f64>,
    pub total_taxes: Option<f64>,
    pub total_ticket_count: Option<i32>,
    pub work_period_informations: Option<String>,
    pub department_ticket_informations: Option<String>,
    pub add_on: Option<String>,
    pub auto_closed: Option<bool>,
    pub external_processed: Option<bool>,
    pub work_period_day: Option<String>,
    pub business_date: Option<String>,
    pub sync_status: String, // PENDING, SYNCING, SYNCED, FAILED
    pub sync_error: Option<String>,
    pub created_at: String,
    pub updated_at: Option<String>,
}
